providers/src/airflow/providers/common/sql/operators/sql.py

```diff
@@ -145,13 +145,25 @@ def __init__(
         self.hook_params = hook_params or {}
         self.retry_on_failure = retry_on_failure
 
+    @classmethod
```

Comment:
When we move the generic_transfer to standard provider, the user might be able to install it and run on Airflow 2 and convert their DAG to use the operators / hooks / etc. from the  "standard" provider. This is happening in order to ease the migration. 

But this means that theoretically, one can install new "standard provider" and old "common.sql" that has no this method installed.  In order to enforce the right dependency - when you extract generic_transfer to standard provider you should add dependency to the standard provider:

```
common-sql >= UPCOMING_VERSION_OF_COMMON_SQL (which is the next minor version as we are adding this method as a new feature)
```

This will make sure that when "standard" provider is installed, common-sql will always be >= NEW_VERSION_OF_COMMON_SQL that contains the new get_hook method - thus the method will be usable by the `generic_transfer` from the standard provider.

Reply:
> NEW_VERSION_OF_COMMON_SQL

I made the standard provider dependent of latest common sql provider, but this won't solve the issue of Airflow 2 with the get_hook method of BaseHook, which resides in the airflow core basecode right?  So I still need the backward compatibility I would say until standard provider depends on Airflow 3.0.0 or higher.  Anyway I created a test case for this not to forget as mentioned above.